## Introduction

Metalloproteins are ubiquitous in nature and are present in all major enzyme families [@doi:10.1021/cr400458x;@doi:10.1093/protein/gzw026].The metals predominantly found in biological systems are the first and second row alkali and earth alkali metals and the first row transition metals such as zinc and copper. <!--The most common metal found in biological systems is Zinc (present in ~10% of deposited structures )ZN ligand in  18889 structures, 187844 total structures deposited in PDB. Considering all Zn-containing structures, the number is unchanged in practice (18923/187844=10%) --> Zinc is the most common transition metal (present in ~10% of deposited structures) and can fulfill both a structural (e.g. in zinc finger proteins) or a catalytic role in up to trinuclear active sites. Zn<sup>2+</sup> is an excellent Lewis acid and is most often found in tetrahedral, pentavalent, or octahedral coordination. About 10 % of all reactions catalyzed by enzymes use zinc as cofactor[@pmid:18604568].

Metalloproteins are well studied because metal cofactors are essential for the function of many proteins and loss of this function is an important cause of diseases [@doi:10.1038/s42256-019-0119-z]. Industrial applications for metalloproteins capitalize on the favorable catalytic properties of the metal ion where the protein environment dictates (stereo)-selectivity [@doi:10.1126/science.aau3744, @doi:10.1021/bi201881p, @doi:10.1039/C5SC01065A; @doi:10.1038/nature17968; @doi:10.1038/s41570-021-00339-5]. To crystallize proteins, metal salts are also often added to the crystallization buffer as they can help in the formation of protein crystals overcoming the enthalpic cost of association of protein surfaces. Metal ion binding sites can be used to engineer protein-protein interactions (PPI) [@doi:10.1038/nchem.1290; @doi:10.1021/ja208015j; @doi:10.1021/ar900273t] and the hypothesis has been put forward that one origin of macromolecular complexity is the superficial binding of metal ions in early single domain proteins [@doi:10.1021/ar900273t].

While simple metal ion binding sites can be rapidly engineered because initial coordination on a protein surface can for example be achieved by creating an i, i+4 di-histidine site on an alpha-helix [@doi:10.1002/anie.202009226] or by placing cysteines in spatial proximity [@doi:10.1038/nchem.1201], the engineering of complex metal ion binding sites e.g. in the protein interior is considerably more difficult [@doi:10.1093/protein/gzw026;@doi:10.1021/ja208015j] as such sites are often supported by a network of hydrogen bonds. A complication for computational design of metalloproteins is the unavailability of good (non-bonded) force fields for zinc and other transition metals that accurately reproduce (e.g. tetrahedral) coordination with the correct coordination distances which renders design using e.g. Rosetta very difficult [@doi:10.1093/protein/gzw026; @doi:10.1021/jacs.0c01329]. In fact, the latest parametrization of the Rosetta energy function (ref2015) [@doi:10.1021/acs.jctc.7b00125] did not refit the parameters for the metal ions which originally are from CHARMM27 with empirically derived Lazaridis-Karplus solvation terms. To adequately treat metal sites in proteins quantum mechanical treatments such as in hybrid quantum mechanics/molecular mechanics (QM/MM) simulations [@doi:10.1021/cr500628b;@doi:10.1021/acs.jcim.1c01109] is needed whose computational cost is prohibitive for regular protein design tasks. QM/MM simulations can however be used to verify coordination chemistry for select candidate proteins [@doi:10.1021/jacs.7b10660]. On the other hand, neural network potentials have been developed for zinc however those require the experimental zinc location as input [@doi:10.3389/fchem.2021.692200].

Many tools exist to predict whether a protein contains metals (e.g. ZincFinder [@doi:10.1186/1471-2105-8-39]), which residues in the protein bind a metal (e.g. IonCom [@doi:10.1093/bioinformatics/btw396], MIB [@doi:10.1021/acs.jcim.6b00407]) and where the metal is bound (AlphaFill [@doi:10.1101/2021.11.26.470110], FindsiteMetal [@doi:10.1002/prot.22913], BioMetAll [@doi:10.1021/acs.jcim.0c00827],MIB [@doi:10.1021/acs.jcim.6b00407] ). The input for these predictors is based on sequence and/or structure information. Sequence-based predictors use pattern recognition to identify the amino acids which might bind a metal [@doi:10.1002/minf.201800169].
Structure-based methods use homology to known structures (MIB, Findsite-metal, AlphaFill) or distance features (BioMetAll) to infer the location of metals. Some tools like Findsite-metal or ZincFinder employ machine learning based approaches such as support vector machines.

Structure based deep learning approaches have been used in the field of protein research for a variety of applications such as protein structure prediction [@doi:10.1038/s41586-021-03819-2, @doi:10.1126/science.abj8754], prediction of identity of masked residues [@doi:10.1186/s12859-017-1702-0; @doi:10.1021/acssynbio.0c00345; @doi:10.1038/s41467-022-28313-9 ]), functional site prediction [@doi:10.1093/bioinformatics/bty813; @doi:10.1038/s41467-021-24070-3], for ranking of docking poses [@doi:10.1038/s41467-021-27396-0; @doi:10.1038/s41592-019-0666-6], prediction of the location of complex organic ligands [@doi:10.1093/bioinformatics/btx350; @doi:10.1093/bioinformatics/bty583; @doi:10.48550/arXiv.2202.05146 ; @doi:10.1038/s41592-019-0666-6], and prediction of effects of mutations for stability and disease [@doi:10.1038/s42256-019-0119-z, @doi:10.1371/journal.pcbi.1008291].
Current state of the art predictors for metal location are MIB [@doi:10.1371/journal.pone.0039252;@doi:10.1021/acs.jcim.6b00407], which combines structural and sequence information in the “Fragment Transformation Method” to search for homologous sites in its database, and BioMetAll [@doi:10.1021/acs.jcim.0c00827], a geometrical predictor based on backbone preorganization. Both methods have significant drawbacks: MIB excludes metal sites with less than 2 coordination partners from its analysis and is limited by the availability of templates in its database. BioMetAll does not use templates but provides many possible locations for putative binding sites on a regular grid. The individual probes in BioMetAll do not have a confidence metric therefore only allowing to rank sites by the number of probes found, which results in a large uncertainty in the position. Both tools suffer from many false positives.  In this work, we present two metal ion location predictors that do not suffer from these drawbacks. The deep learning based Metal3D predictor operates on a voxelized representation of the protein environment and predicts a per residue metal density that can be averaged to get a smooth metal probability density over the whole protein. The distance based predictor Metal1D predicts the location of metals using distances mined from the protein data bank (PDB) directly predicting coordinates of the putative metal binding site. These tools pave the way to perform in silico design of metal ion binding sites without relying on predefined geometrical rules or expensive quantum mechanical calculations. 
<!--This will also allow to exploit the catalytic diversity of metal ions to design enzymes for new-to-nature reactivities more easily.-->