## Introduction

Metalloproteins are ubiquitous in nature and are present in all major enzyme families [@doi:10.1021/cr400458x;@doi:10.1093/protein/gzw026]. The metals typically found in biological systems are the first and second row alkali and earth alkali metals and the first row transition metals such as zinc and copper. <!--The most common metal found in biological systems is Zinc (present in ~10% of deposited structures )ZN ligand in  18889 structures, 187844 total structures deposited in PDB. Considering all Zn-containing structures, the number is unchanged in practice (18923/187844=10%) --> Zinc is the most common transition metal (present in ~10% of deposited structures) and can fulfill both a structural role (e.g in Zinc finger proteins) or a catalytic role in up to trinuclear active sites. Zn<sup>2+</sup> is an excellent Lewis acid and is most often found in tetrahedral, pentavalent or octahedral coordination. About 10 % of all reactions catalyzed by enzymes use zinc 

Metalloproteins are well studied because metal cofactors are essential for the function of many proteins.  Loss of this function is an important cause of diseases [@doi:10.1038/s42256-019-0119-z]. Industrial applications for metalloproteins result from the favorable catalytic properties of the metal ion with the tunable protein environment dictating (stereo)-selectivity [@doi:10.1126/science.aau3744, @doi:10.1021/bi201881p, @doi:10.1039/C5SC01065A]. To crystallize proteins, metal salts are also often added to the crystallization buffer as they can help in the formation of protein crystals overcoming the enthalpic cost of association of protein surfaces.  Metal ion binding sites can be used to rapidly engineer protein-protein interactions (PPI) [@doi:10.1038/nchem.1290; @doi:10.1021/ja208015j; @doi:10.1021/ar900273t] and the hypothesis has been put forward that one origin of macromolecular complexity is the superficial binding of metal ions in early single domain proteins [@doi:10.1021/ar900273t]. Many metal ion binding sites are still found at interfaces in modern proteins. 

While simple metal ion binding sites can be rapidly engineered because initial coordination on a protein surface can for example be achieved by creating an i, i+4 di-histidine site [@doi:10.1002/anie.202009226] or by placing cysteines in spatial proximity [@doi:10.1038/nchem.1201] the engineering of complex metal ion binding sites is considerably more difficult [@doi:10.1093/protein/gzw026;@doi:10.1021/ja208015j] as such sites are often supported by a network of hydrogen bonds. A complication for computational design of metalloproteins is the unavailability of good non-bonded force fields for zinc and other transition metals that accurately reproduce tetrahedral coordinatio with the correct coordination distances which renders design e.g using Rosetta very difficult [@doi:10.1093/protein/gzw026; @doi:10.1021/jacs.0c01329]. The latest parametrization of the Rosetta energy function (ref2015) [@doi:10.1021/acs.jctc.7b00125] did not refit the parameters for the metal ions which originally are from CHARMM27 with the Lazaridis-Karplus solvation terms being a wild guess [sic]. To treat metal sites in proteins adequately quantum mechanical treatment such as in hybrid quantum mechanics/molecular mechanics simulations [@doi:10.1021/cr500628b] is needed whose computational cost is prohibitive for regular protein design tasks. QM/MM simulations can however be used to verify coordination chemistry for select candidate proteins [@doi:10.1021/jacs.7b10660]. 

Many tools exists to predict whether a protein contains metals (e.g ZincFinder [@doi:10.1186/1471-2105-8-39]), which residues in the protein bind a metal (e.g IonCom [@doi:10.1093/bioinformatics/btw396], MIB [@doi:10.1021/acs.jcim.6b00407]) and where the metal is bound (AlphaFill [@doi:10.1101/2021.11.26.470110], FindsiteMetal [@doi:10.1002/prot.22913], BioMetAll [@doi:10.1021/acs.jcim.0c00827] ). The input for these predictors is based on sequence and/or structure. Sequence-based predictors use pattern recognition to identify the amino acids which might bind a metal based e.g on a multiple sequence alignment.
On the other hand, structure-based methods are based on the geometrical 3D structure, either using homology to known structures (MIB, Findsite-metal, AlphaFill) or distance features (BioMetAll) to infer the location of metals. Some tools like Findsite-metal employ machine learning based approaches. 
Modern deep learning approaches have the advantage that they do not rely on careful feature extraction/engineering to extract relevant features. Structure based deep learning based approaches have been used in the field of protein research for a variety of applications such as protein structure prediction [@doi:10.1038/s41586-021-03819-2, @doi:10.1126/science.abj8754], prediction of identity of masked residues [@doi:10.1186/s12859-017-1702-0; @doi:10.1021/acssynbio.0c00345; @doi:10.1038/s41467-022-28313-9 ]), functional site prediction [@doi:10.1093/bioinformatics/bty813; @doi:10.1038/s41467-021-24070-3], for ranking of docking poses [@doi:10.1038/s41467-021-27396-0; @doi:10.1038/s41592-019-0666-6], prediction of the location of complex organic ligands [@doi:10.1093/bioinformatics/btx350; @doi:10.1093/bioinformatics/bty583; @doi:10.48550/arXiv.2202.05146 ; @doi:10.1038/s41592-019-0666-6] and prediction of effects of mutations for stability and disease [@doi:10.1038/s42256-019-0119-z, @doi:10.1371/journal.pcbi.1008291]. Most of those models represent the protein as voxels und apply 3D convolutional neural networks (CNN). Other architectures that treat the protein as surfaces or graphs often are constrained to use coarse grained sidechain information due to the computational cost. 

Current state of the art predictors for metal-binding sites are MIB [@doi:10.1371/journal.pone.0039252;@doi:10.1021/acs.jcim.6b00407], which combines structural and sequence information in the "Fragment Transformation Method" to search for homologous sites in its database, and BioMetAll [@doi:10.1021/acs.jcim.0c00827], a geometrical predictor based on backbone preorganization. Both methods have significant drawbacks: MIB excluded metal sites with less than 2 coordination partners from its analysis and is limited by the availability of templates in its databank. BioMetAll does not use templates but provides many possible locations for putative binding sites on a regular grid. The individual probes in BioMetAll do not have a confidence metric therefore only allowing to rank sites by the number of probes found which results in a large uncertainity in the position. Both tools suffer from many false positives. <!--  The reported accuracy and sensitivity for the MIB predictor are 94.6% and 64.7%, respectively (average values on different metal ions, for Zn<sup>2+</sup> in particular the corresponding values are 94.8% and	71.1%). For the BioMetAll predictor, the same quantities are not available, but a benchmark on 53 two-histidine one-carboxylate motifs resulted in an average distance between the predicted and the experimental site of the metal of (0.56±0.19)Å Put cluster center instead of one of the probe. These values refer to the performances assessed in the original publications, but are not straightforward to compare and extend to the general case. For this reason, in this work we also performed a robust and fair comparison of our two novel methods with MIB and BioMetAll predictors. -->
In this work, we present two metal ion location predictors that do not suffer from these drawbacks. The deep learning based Metal3D predictor operates on a voxelized representation of a protein environment and predicts a per residue metal density that can be averaged to get a smooth metal probability density over the whole protein. The distance based predictor Metal1D predicts the location of metals using distances mined from the protein databank predicting coordinates directly. 
These tools pave the way to perform in silico design of metal ion binding sites without relying on predefined geometrical rules or expensive quantum mechanical calculations. This will allow to exploit the catalytic diversity of metal ions to design enzymes for new-to-nature reactivities more easily. 

<!-- 

many biological functions [Percora review]? 
As oneof the most prevalent transition metal cofactors in
biological systems, it plays structural, signaling, and regulatory
roles and is found in all six classes of enzymes (most commonly
hydrolases



Understanding where metals bind in biology is related to health [@doi:10.1038/s42256-019-0119-z], biocatalyis [Kuhlman, Hilvert] and PPIs [Tezcan]. 

Starting with pioneering studies in the 1990 s,[@doi:10.1126/science.8346440;@doi:10.1146/annurev.biochem.68.1.779]there have been notable successes inthe de novo design of functional metalloproteins, which arepredominantly based on four-helix bundle anda-helicalcoiled-coiled motifs with readily parametrizable structure [@doi:10.1002/anie.202009226]. 


Zinc exists as a redox-inert Zn(II) cation with an electron configuration of [Ar]3d10
These properties with the lack of ligand field effects make zinc an excellent metal for different coordination numbers and binding geometries in different biological systems.
 Zinc can be found as an active site metal (cofactor) in all six IUBMB enzyme classes
### Deep learning on proteins
Torng/Shroff 3DCNN bio stuff 
Ananad DeepRank
Correia surface studies Nat Methods 


### Enzymes

### Interfaces
It has been hypothesized that some modern metalloproteins may have emerged through the metal-nucleated oligomerization of small peptides or protein domains, followed by the evolution of the resulting assemblies into stable, functional architecture.


Metal-Templated Interface Redesign (MeTIR) ). These strategies, inspired by both the proposed evolutionary roles of metals and their prevalence in natural PPIs, take advantage of the favorable properties of metal coordination (bonding strength, directionality, and reversibility) to guide protein self-assembly with minimal design and engineering

In order to circumvent the complexity of constructing extensive noncovalent interfaces, which are typically involved in natural PPIs

### Existing approaches

Computational predictors of metal-binding sites built on sequence analyses are mostly based on scanning the sequence of a target protein to identify those regions where amino acid patterns match a metal-binding site fingerprint. For zinc binding sites often two histidine spaced by one residue which allows to readily detect the motif are easy to detect. These predictors yield the identites of the coordinating residues. 

Structural detectors often used distance features to identify sites based on statistical mining in the protein databank.  Predictors trained like this can identify highly preorganized motifs (e.g 4x Cys in close spatial proximity) but are often not very good at identifying weakly preorganized motifs. Current state of the art predictors (MIB, BioMetAll) use fragments/homology to predict the location of the metal or backbone preorganization predicting an approximate position of the metal that is less sensitive to the exact side chain geometry thus affording higher sensitivity to detect metal sites(BioMetAll). MIB [@doi:10.1021/acs.jcim.6b00407] uses the fragmentation transformation method to search for homologus sites in its database


In our work we develop two new predictors primarily intended for zinc binding sites - Metal1D and Metal3D that are more accurate and sensitive than existing approaches in predicting metal ion binding sites. We evaluate their capability with respect to  -->

