## Results

A dataset of experimental high resolution crystal structures (2085 structures/266 000 environments) containing zinc sites was used for training of the geometric predictor Metal1D and the deep learning predictor Metal3D  (Figure @fig:method). For training, we used the crystal environment including crystal contacts. For predictions, the biological assembly was used. 

![ **A** Training of Metal3D and Metal1D is based on experimental Zn<sup>2+</sup> sites. Metal1D extracts coordination environments from LINK records, Metal3D is a fully convolutional 3DCNN trained to predict the metal density from voxelized protein environments. **B** In inference mode Metal3D predicts the location of a metal ion by computing per residue metal densities and then averaging them to obtain a global metal density for the input proteins. The ions can then be placed using the weighted average of voxels above a cutoff. For Metal1D all residues in the protein are scanned for compatibility with the probability map. Metals are placed at the geometric center of residues with high scores according to the probability map. A final ranking of sites is obtained using the probability map. 
](images/Metal3D_Metal1D_method.png){#fig:method}

### Metal3D

Metal3D takes a protein structure and a set of residues as input, voxelizes the environment around each of the residues and predicts the per residue metal density. The predicted per residue densities (within a 16 x 16 x 16 Å<sup>3</sup> volume)  can then be averaged to yield a zinc density for the whole protein. At high probability cutoffs the predicted metal densities are spherical (Figure @fig:2cbaprobabilities E), at low probability cutoffs the predicted densities are non-regular (Figure @fig:2cbaprobabilities A).

![Probability evolution in HCA2 for different probability cutoffs A) p=0.1 B) p=0.2 C) p=0.3 D) p=0.4 E) p= 0.75. <a href="https://lcbc-epfl.github.io/metal-site-prediction/interactive/index.html">Link to interactive visualization</a>](images/2CBA_probabilities.png){#fig:2cbaprobabilities}


We evaluated the quality of the metal densities generated by the model with the discretized Jaccard similarity (Figure @fig:jaccard) for all environments in the test set. We noticed that at the edges of the residue-centered probe boxes often spurious density is predicted wherefore we evaluated the similarity of the test set metal density and the predicted metal probability density taking into account a smaller box with truncated edges. Figure  @fig:jaccard shows that the similarity of the boxes does not depend much on the probability cutoff chosen with higher cutoffs yielding slightly higher discretized Jaccard similarity values (0.02 - 0.04 difference between p=0.5 and p=0.9). Reducing the size of the analyzed boxes (i.e trimming of the edges) increases the Jaccard similarity from ≈ 0.64 to 0.88 showing that the metal density in the center of the box is indeed more accurate than the density at the edges.

Metal3D is available as self-contained notebook on GoogleColab and on Huggingface Spaces:

<iframe frameborder="0" width="100%" height="900px" title="Gradio Metal3D" src="https://hf.space/embed/simonduerr/metal3d/+?__theme=light"></iframe>
<!-- <iframe frameborder="0" width="100%" height="900px" title="Gradio Metal3D" src="https://15420.gradio.app"></iframe> -->

<!--<script src="https://gist.github.com/lzhou1110/2a30a81cb8c175514ed627bc18016774.js"></script>-->

### Metal1D
The statistical analysis for the geometric predictor uses the LINK records present in deposited PDB structures. A probability map for all zinc coordination motifs was extracted from all training structures. The mean coordination distance in the training set was found to be  2.2 ± 0.2 Å, and the default search radius was set to 5.5 Å (Table @tbl:metal1dreferencepoints). In total 208 different environments with more than 5 different proteins (at 30 % sequence identity) were identified. 
Metal1D is available as self-contained notebook on Google Colab


<!--<script src="https://gist.github.com/lzhou1110/2a30a81cb8c175514ed627bc18016774.js"></script>-->



### Comparison of Metal1D, Metal3D, MIB and BioMetAll

Existing metal ion predictors can be subdivided into two categories: binding site predictors and binding location predictors. The former identify only the residues binding the ion, the latter predict the coordinates of the metal ion itself. Both Metal1D and Metal3D can predict the coordinates of putative binding sites. We therefore assessed their performance by comparing to recent binding location predictors, namely  BioMetAll [@doi:10.1021/acs.jcim.0c00827]  and MIB [@doi:10.1021/acs.jcim.6b00407].  The main tuning parameter of MIB is the template similarity t,  with higher values requiring higher similarity of the templates available for the search in structurally homologous metalloproteins. BioMetAll on the other hand was calibrated on available protein structures and places probes on a regular grid at all sites where the criteria for metal binding are fulfilled. The main adjustable parameter for BioMetAll is the cluster cutoff c, which indicates how many probes in reference to the largest cluster a specific cluster has. We used the recommended cutoff of 0.5 requiring all chosen clusters to have at least 50% of the probes of the most populous cluster and used the cluster center to compute distances.

We first investigated the potential of all tools to detect the location of a zinc ion binding site in a binary fashion (zinc site or no zinc site). We defined a correctly identified binding site (true positive, TP) as a prediction within 5 Å of an experimental zinc site. In case a tool predicted no metal within the 5 Å radius, we counted this site as false negative (FN). False positive (FP) predictions, i.e sites where a metal was placed spuriously, were clustered in a 5 Å radius and counted once per cluster. All tools were assessed against the held out test biological assemblies for Metal3D and Metal1D. When the performance of MIB (t=1.25) and BioMetAll is compared against Metal3D with probability cutoff p=0.75 we find that Metal3D identifies more sites (85) than MIB (78) or BioMetAll (75) with a much lower number of false positives (Figure @fig:comparison).
MIB predicts 180 false positive sites, BioMetAll 134 sites whereas Metal3D only predicts 9 false positive sites at the p=0.75 cutoff. Metal1D (t=0.5) offers similar detection capabilities (78 sites detected) with a lower number of false positives (47) compared to MIB and BioMetAll. 
As about 1/3 of the zinc binding sites in the PDB are believed to be artifacts [@doi:10.1021/ic401072d] we removed 56 sites from the list of zinc sites in the test set (189 total) that had less than 2 unique protein ligands within 2.8 Å of the experimental zinc location. The amount of correct predictions in this reduced set is almost unchanged for all tools (Figure @fig:comparison) indicating that most tools correctly predict sites if they have 2 or more protein ligands. For Metal3D at p=0.75 and p=0.9 as well as MIB with t=1.9 all sites that are correctly predicted to contain a metal are sites with more than 2 protein ligands. The number of false negatives is reduced for all tools by about 50 sites indicating that most tools do not predict these crystallographic artifacts that might depend on additional coordinating residues from an adjacent molecule in the crystal. Of all tools, Metal3D has the least false positives (1 FP at p=0.9) and the highest number of detected sites (110 at p=0.25). The single false positive at p=0.9 does not contain a zinc ion but is a calcium binding site with three aspartates and one backbone carbonyl ligand (Figure @fig:4jjjFP).

![Comparison of Metal1D, Metal3D, BioMetAll and MIB on the test set held out from training of Metal1D and Metal3D. Predicted sites are counted as true positives (TP) if they are within 5 Å of a true metal location and as false negatives (FN) otherwise. False positive (FP) probes are clustered and counted once per cluster. *For MIB we used 2 structures less because the server did not accept these structures.](images/all_2coord_metal_1d_3d_min_biometall_comparison_lowres.jpg){#fig:comparison}


After assessment of how many sites the tools predict, another crucial metric is the spatial precision of the predictions. For the correctly identified sites (TP) we measured the mean absolute distance (MAD) between experimental and predicted position (Figure @fig:selectivity-mad). The MAD for Metal3D at p=0.9 is 0.70 ± 0.64 Å and 0.73 ± 0.65 Å at p=0.25 indicating that low confidence predictions are still accurately placed inside the protein. The median MAD of predictions for Metal3D at p=0.9 is 0.52 Å indicating that for half of the predictions the model predicts at or better than the grid resolution of 0.5 Å.

BioMetAll is not very precise with an MAD for correctly identified sites of 2.80 ± 1.30 Å. BioMetAll predicts many possible locations per cluster with some of them much closer to the experimental metal binding site than the cluster center. However, it does not provide any ranking of the probes within a cluster and therefore the cluster center was used for the distance calculation. Metal1D (MAD 2.06 ± 1.33 Å) which identifies more sites than BioMetAll is also more precise than BioMetAll. MIB t=1.9 detects sites with high precision (MAD 0.77 ± 1.09 Å) but it relies on the existence of homologous sites to align the found sites.

![**A** Mean absolute deviation (MAD) of predicted zinc ion locations using Metal1D, Metal3D, BioMetAll and MIB on the test set used to train Metal1D and Metal3D for all correctly identified (TP) sites. n is the number of sites predicted by the tool. *For MIB we used 2 structures less because the server did not accept these structures. For each tool the whisker plot indicates the median (white dot) and the first quartiles (black box). **B** Recall for the zinc test set and 25 randomly drawn structures for other transition, alkali and earth-alkali metal ions for Metal3D using p=0.5 as cutoff. ](images/Figure3_selectivity_mad.png){#fig:selectivity-mad}

### Selectivity for other metals

In spite of being exclusively trained on zinc, the performance of Metal1D and Metal3D was also assessed on sodium (Na+, PDB code NA), potassium (K+, PDB code K), calcium (Ca2+, PDB code CA), magnesium (Mg2+, PDB code MG), and various transition metals (Fe2+, Fe3+, Co2+, Cu2+, Cu+, Mn2+ with corresponding PDB codes FE2, FE, CO, CU, CU1, MN, respectively) from 25 randomly drawn structures from the clustered PDB at 30% identity. Only sites with at least 2 protein ligands were used for the analysis to exclude crystallographic artifacts. Figure 7 shows that recall for Metal3D is high for all transition metals, meaning that the model correctly finds most sites even though it was only trained on zinc. For the alkali and earth alkali metals recall is much lower as the model only finds some sites. The mean probability for found zinc structures (ZN p= 0.95 ± 0.10) in the test set is higher than for the other transition metals (Figure @fig:selectivity-probability-metal3d) and significantly higher than that for alkali metals (NA p=0.61 ± 0.10, K p=0.79 ± 0.16) while the probability for the earth alkali metals is slightly higher with MG (p=0.77 ± 0.16)  similar to CA (p=0.73 ± 0.16). The MAD for each found metal site is again lowest for zinc (0.56 ± 0.59 Å). The MAD for the found sodium (n=2) and potassium (n=5) sites are as low as for the other transition metals. The only metal with significantly higher MAD (1.45 ± 0.93 Å) is CU1 (Figure @fig:selectivity-distance-metal3d).

The only two structures where a sodium is detected by Metal3D  (2OKQ[@doi:10.2210/pdb2OKQ/pdb], 6KFN[@doi:10.2210/pdb6KFN/pdb]) have at least 2 side chain coordinating ligand atoms and only one backbone (2OKQ) or no backbone ligand atom (6KFN). Canonical sodium binding sites e.g. such as in PDB 4I0W [@doi:10.2210/pdb4I0W/pdb] with two coordinating backbone carbonyl oxygen atoms and one asparagine side chain have probabilities around 5 % and are basically indistinguishable from background noise of the model.
For Metal1D overall recall is lower with similar differences in the detection of main group metals versus transition metals (Figures @fig:selectivity-metal1d and  @fig:selectivity-distance-metal1d).




### Applications

After having evaluated the accuracy of Metal3D on held out test structures we also investigated possible uses in downstream applications such as protein function annotation and protein design. 

#### Alpha Fold 

AlphaFold2 often predicts side chains in metal ion binding sites in the holo conformation [@doi:10.1038/s41586-021-03819-2]. Tools like AlphaFill [@doi:10.1101/2021.11.26.470110]use structural homology to transplant metals from similar PDB structures to the predicted structure. Metal3D does not require explicit homology based on sequence alignment like AlphaFill so it is potentially suited to annotate the dark proteome that is now accessible from the AlphaFold database with zinc ions. Metal3D identifies both the structural and catalytic sites for the example (PDB 3RZV[@doi:10.2210/pdb3RZV/pdb] , Figure @fig:alphafold A) used in ref [@doi:10.1101/2021.11.26.470110] with high probability (p=0.99) even though one of the sites in the AlphaFold model is slightly disordered with one of the binding residues in the solvent facing conformation (D309). 
The distances between predicted and modeled metal locations for Metal3D are 0.22 Å and 0.37 Å, for AlphaFill they are 0.21 Å and 0.41 Å. AlphaFill uses a 25% sequence identity cutoff which can be problematic for certain proteins with no structurally characterized homologues. For human Palmitoyltransferase ZDHHC23 (Uniprot Q8IYP9) a high confidence AlphaFold2 prediction exists but AlphaFill cannot place the zinc ions because the sequence overlap is 24% to the closest PDB structure (PDB 6BMS [@doi:10.1126/science.aao6326]), just below the 25% cutoff. For the identical site in another human Palmitoyltransferase ZDHHC15 (Uniprot Q96MV8) AlphaFill is able to place the metal because of higher sequence identity to 6BMS (64%) (Figure @fig:alphafold B). Metal3D in both cases is able to place the metal with high confidence (MAD 0.75 Å for site 1 and 0.48 Å for site 2, p>0.99 ) based on the single input structure alone. 





![**A** Predicted metal binding sites (a and b) from Metal3D, respectively AlphaFill compared to the experimentally found zinc positions for Uniprot O95630. Metal3D places the metal with high accuracy even if sidechains are not perfectly predicted by AlphaFold for site 1 **B** Palmitoyltransferase ZDHHC23 (Uniprot Q8IYP9)  and ZDHHC15 (Uniprot Q96MV8). AlphaFill can only place the metal for ZDHHC15 because sequence identity for ZDHHC23 is 24 %. Probability isosurfaces from Metal3D for both structures at p=0.6, colored in gray.](images/3rzv_6bsm_AF2_alphafill_metal3d.png){#fig:alphafold}



#### Metalloprotein engineering

Human carbonic anhydrase II (HCA2) is a well studied metalloenzyme with a rich amount of mutational data available. For  the crystal structure of the wildtype enzyme (PDB 2CBA [@doi:10.2210/pdb2cba/pdb]),  Metal3D recapitulates the location of the active site metal with a RMSD to the true metal location of 0.21 Å with a probability of p=0.99. At lower probability cutoffs (p<0.4) the probability map indicates further putative metal ion binding sites with interactions mediated by surface residues (e.g. H36, D110, p=0.4) (Figure @fig:2cbaprobabilities). 


<iframe src="interactive/index.html" title="2CBA probabilities" width="100%" height="900px" frameborder="0"></iframe> 


To investigate the capabilities for protein engineering we used mutational data for first and second shell mutants of the active site residues in HCA2 with corresponding K<sub>d</sub> values from a colorimetric assay [@pmid:3887984]. For most mutants no crystal structures are available so we used the structure builder in the EVOLVE package <!--REF-->  to choose the most favorable rotamer for each single point mutation based on the EVOLVE-ddg energy function <!--REF-->  with explicit zinc present (modeled using a dummy atom  approach [@pmid:11106157]). he analysis was run for every single mutant and the resulting probability maps were analyzed. For the analysis we used the maximum predicted probability as a surrogate to estimate relative changes in K<sub>d</sub>. For mutants that decrease zinc binding drastically we observe a drop in probability (Figure @fig:hca-kd).The lowest probability mutants are H119N and H119Q with p=0.23 and 0.38. The mutant with the largest loss in zinc affinity H94A has a zinc binding probability of p=0.6. Conservative changes to the primary coordination motif (e.g. H &rarr; C) reduce the predicted probability by 10 - 30 %. For second shell mutants the influence of the mutations is less drastic with only minor changes in the predicted probabilities. 

![Experimentally measured K<sub>d</sub> values [@doi:10.1021/bi00255a003;@doi:10.1021/ja00079a046;@doi:10.1021/bi00089a005;@doi:10.1073/pnas.92.11.5017;@doi:10.1021/bi9526692] for 1<sup>st</sup> and 2<sup>nd</sup> shell active site mutants of HCA2 and predicted max probability for zinc using Metal3D.](images/kd_vs_p_nolog_newmethod_newmodel_0.5.png){#fig:hca-kd}










<!--Many metal ion predictors exist that can be subdivided into two categories: binding site predictors and binding location predictors. The former label only the residues binding the ion, the latter predict the coordinates of the ion. Both Metal1D and Metal3D can predict the coordinate of putative binding sites. We therefore assessed the performance comparing to recent predictors that also directly predict the location of the metal (BioMetAll, MIB). Other tools such Findsite-metal are no longer available online. 
MIB uses a fragment method to identify homologous binding sites to the motifs it finds in a given structure and will extract the location of the metal from the homologous structures in its database aligning it to the current site. The main parameter of MIB is the template similarity `t` with higher values requiring higher similarity of the templates available for the search. 
BioMetAll was calibrated on available protein structures and places probes on a regular grid at all sites where the criteria for metal binding are fulfilled. Probes are associated to a cluster of amino acids.  For each cluster of probes also a geometric center of the probes is given which we used to assess performance in this work as there is no individual ranking of the probes given by the program. The main parameter for BioMetAll is the cluster cutoff `c` which indicates how many probes in reference to the largest cluster a specific cluster has. We used the recommended cutoff of 0.5 requiring all chosen clusters to have at least 50% of the probes of the most populous cluster.  
We first investigated the potential of all tools to detect a metal ion binding site. We defined a correctly identified binding site as a prediction within 5 Å of an experimental zinc site. In case a tool predicted no metal within the 5 Å radius we counted this site as false negative. False positive predictions, i.e sites where a metal was placed spuriously, were clustered and counted once per cluster.
When the performance of MIB (`t=1.25`) and BioMetAll is compared against Metal3D with probability cutoff p=0.75 we find that Metal3D identifies more sites (85) than MIB (78) or BioMetAll (75) with a much lower number of false positives (Figure @fig:comparison). MIB predicts 180 false positive sites, BioMetAll 134 sites whereas Metal3D only predicts 9 false positive sites at the p=0.75 cutoff. Metal1D (`t=0.5`) offers similar detection capabilities (78 sites detected) with a lower number of false positives (47) compared to MIB and BioMetAll. As about 1/3 of zinc binding sites in the PDB are artifacts [@doi:10.1021/ic401072d] we removed 56 sites from the list of zincs in the test set (189 total) that had less than 2 unique protein ligands within 2.8 Å of the experimental zinc. The amount of correct predictions in this reduced set is almost unchanged for all tools (Figure @fig:comparison) indicating that most tools correctly predict sites only if they have 2 or more protein ligands. For Metal3D at p=0.75 and p=0.9 as well as MIB `t=1.9` all sites that are correctly predicted to contain a metal are sites with more than 2 protein ligands. The number of false negatives is reduced for all tools by about 50 sites indicating that most tools do not predict these crystallographic artefacts that might depend on additional coordinating residues from an adjacent molecule in the crystal. 
Of all tools, Metal3D has the least false positives (1 FP at p=0.9) and the highest number of detected sites (110 at p=0.25). The single false positive at p=0.9 is a calcium binding site with three aspartates and one backbone carbonyl ligand (Figure @fig:4jjjFP).


After assessment of how many sites the tools predict, another crucial metric is the spatial precision of the predictions. For the correctly identified sites we measured the mean absolute distance (MAD) between experimental and predicted position (Figure @fig:selectivity-mad). The MAD for Metal3D p=0.9 is 0.70 ± 0.64 Å and 0.73 ± 0.65 Å at p=0.25 indicating that low confidence predictions are still accurate. The median of predictions for Metal3D p=0.9 is 0.52 Å indicating that for half of the predictions the model predicts at or better than the grid resolution of 0.5 Å. 

BioMetAll is not very precise with MAD for correctly identified sites of 2.8 ± 1.3 Å. BioMetAll predicts many possible locations per cluster with some of them much closer to the experimental metal than the center. However, within a cluster the probes are not ranked wherefore the cluster center was used for the distance calculation.  Metal1D (MAD 2.06 ± 1.33 Å) which identifies more sites than BioMetAll is also slightly more precise than BioMetAll. MIB `t=1.9` detects sites with high precision (MAD 0.77 ± 1.09 Å) because it can use homologous sites to align the found sites on the current site. 




### Selectivity for other metals

![Recall for zinc test set and a 25 randomly drawn structures for other transition, alkali and earth-alkali metals for Metal3D using p=0.5 as cutoff](images/model_0.5metal3D_metal_selectivity_0.5.jpg){#fig:selectivity-mad}

Trained exclusively on zinc, the performance of Metal1D and Metal3D was assessed on sodium (NA), potassium(K), calcium (CA), magnesium (MG), and various transition metals (PDB codes FE2, FE, CO, CU, CU1, MN, NI) from 25 randomly drawn structures with one or more metal sites. Only sites with a least 2 protein ligands were used for the analysis to exclude crystallographic artefacts. Figure @fig:selectivity-mad shows that recall for Metal3D is high for all transition metals, meaning that the model correctly finds most sites even though it was only trained on zinc. For the alkali and earth alkali metals recall is much lower as the model only finds some sites. 
The mean probability for found zinc structures (ZN p= 0.95 ± 0.10) in the test set is higher than for the other transition metals (Figure @fig:selectivity-probability-metal3d) and significantly higher than for the alkali metals (NA p=0.61 ± 0.10, K p=0.79 ± 0.16). The probability for MG (p=0.77 ± 0.16) is similar to CA (p=0.73 ± 0.16). The MAD for each found metal site is again lowest for zinc (0.56 ± 0.59 Å). The MAD for the found sodium (n=2) and potassium (n=5) sites are as low as for the other transition metals. The only metal with significantly higher MAD (1.45 ± 0.93 Å) is CU1 (Figure @fig:selectivity-distance-metal3d).

The only two structures where a sodium is detected (2OKQ[@doi:10.2210/pdb2OKQ/pdb], 6KFN[@doi:10.2210/pdb6KFN/pdb]) have at least 2 sidechain ligand atoms and only one backbone (2OKQ) or no backbone ligand atom coordinating it (6KFN). Canonical sodium binding sites e.g such as in PDB 4I0W [@doi:10.2210/pdb4I0W/pdb] with two backbone carbonyl and one asparagine sidechain have probabilities around 5 % and are basically indistinguishable from background noise. 

For Metal1D overall recall is lower with similar differences in the detection of main group metals versus transition metals  (Supplemental figures  @fig:selectivity-metal1d and  @fig:selectivity-distance-metal1d).
-->


<!-- #### Hidden/transient metalsites

Metallodrugs are in important class of drugs that rely on binding inhibitors to a protein (or DNA). Metal3D can be used to screen the hidden metalloproteome by finding transient metal ion binding sites. 

The site where Rapta binds is detected with p=0.3 but in a high resolution structure without (1KX4) there is a salt bridge with a lysine that might occlude metal detection. One could weight by the rotamer/do MD simulation e.g similar as for cryptic pockets. 

![RAPTA (Ru-arene-phosphaadamantane) inhibitor bound to Nucleosome core particle. Metal3D identifies the two Ruthenium sites with low probabilities.](images/5xf6_rapta_vis.png){#fig:drugdesign} -->

