
## Discussion


Metal3D predicts the probability distribution of zinc in protein crystal structures based on a model trained on natural protein environments. The model is trained similar to image segmentation and is trained on binary output i.e if a voxel contains metal or not. By limiting the training samples to high resolution crystal structures (2.5 A) we achieve good results with limited noise after grid averaging. The use of high resolution structures is necessary because at resolution greater than the average zinc coordination distance (2.2A) the distance deviations of the coordination partners with the metal noticeably increases [@doi:10.1021/ic401072d]. 
In contrast to the currently available tools we did not filter the training examples for certain coordination requirements (i.e only sites with at least 3 protein ligands). This allows the model to also predict sites with partial coordination. This is necessary for protein design software to carry out iterative residue sampling starting from a non-metal binding protein and design first a nucleus binding site and then fullfil the remaining coordination requirements. Methods such as MIB which rely on homology therefore perform worse on such weak binding sites because reducing the quality of the available templates by including 1- or 2-coordinate sites would yield many false positives (similar to including less homologous structures for the template search).  Metal3D also explicitly knows about the environment in the vicinity of a putative site and therefore can use the local context to better triage actual metal ion binding sites from other interactions. 
The majority of sites in the training set are well coordinated sites and during training the model can relate the frequency of different metalion binding motifs in the training set with the probability of metal presence. The fraction of artifactutal zinc binding site in the protein data bank is estimated to be about 1/3[@doi:10.1021/ic401072d] with the zinc ions aiding crystal formation. As we train on biological assemblies some of those binding sites might be located at the surface of the assembly whereas in the assymetric unit of the crystal they might form part of a crystal contact. For the model these half-sites can be seen as data augmentation strategy with noisy incomplete sites. 

Empty coordination sites of metals are filled with water molecules in case there is no other exogenous ligand with higher affinity present (e.g a thiol). Our models do not consider water molecules in the input as the quality of water molecules in the protein databank is not very consistent and in protein design algorithms explicit water molecules are usually implcitily modeled. Common water placement algorithms to place deep waters are either computationally expensive or do not consider the location for metal ions (Dowser++ [@doi:10.1002/prot.25081]). For Metal3D the model inputs for the total density however encode an implicit water function where all empty space can be interpreted as the solvent. <!--For Metal1D sites with water ligands were excluded in the training which might explain part of its worse performance with respect to Metal3D. -->

This work is the first to report a modern deep learning based model destined for metal-protein docking. In the more general space of protein-ligand docking a variety of architectures and representations have been used. 3D CNN based approaches such as LigVoxel and DeepSite commonly use a resolution of 1 Å and similar input features as our model to predict the ligand density. However, predicting the density of a multi-nuclear ligand is more complex than predicting the density of mononuclear metal ions. We therefore did not deem it necessary to include a conditioning on how many metal ions are present in the box and rather chose to reflect this in the training data where the model needs to learn that only about half of the environments it sees contain one or more metals. This choice is validating by the fact that outputed probability densities at sufficiently high probability cutoff are sphereical with their radius approximately matching the VDW radius of Zinc.  Mesh convolutional neural networks trained on  protein surface representation [@doi:10.1038/s41592-019-0666-6] also have been used to predict the location and identity of protein ligands but this approach can only label the regions of the surface that bind the metal ion and the algorithm would need to be modifed to give the exact location of the metal. Some metal ion binding sites are also heavily buried inside proteins as they mediate structural stability rendering them inaccessible to a surface based approach. The most recent approaches such as EquiDock use equivariant neural networks such as En-Transformer [@doi:10.48550/arXiv.2102.09844] to predict binding keypoints (defined as 1/2 distance between the C$_{alpha}$ of the binding residue and a ligand atom). Explicit sidechains are still too expensive for such models. In addition the current state of the art architecture achieves a mean RMSD for the ligand of <2 Å only for 25% of their testset. Our model in contrast has a a mean RMSD of <1 Å over the whole testset which remains almost constant no matter the chosen probability cutoff. The lack of accuracy of the equivariant approaches renders them unsuitable for design of complex metal ion binding sites supported by an intricate network of hydrogen bonds that need to be positioned with sub-angstrom accuracy. Our model in contrast is computationally more expensive and less dataefficient due to the need to voxelize the input and provide different rotations of the input environment. 
In contrast to sequence based models [@doi:10.1101/2021.12.22.473759;@doi:10.1038/s41598-018-34533-1] our model is more data efficient and better suited to be integrated into structure based design workflows such as Rosetta or EVOLVE. Sequence based approaches can only use coevolution signals to infer residues in spatial proximity that can bind a metal. This is easy for sites that contain a HxH motif such as found in HCA2 but might be considerably more difficult when it comes to ranking different sites (e.g histidine versus cysteine) where subangstrom level precision is needed to perform a successful ranking. 

Protein design using 3D convolutional neural nets trained on residue identity has been successfully demonstrated and we anticipate that our model could be seamlessly integrated into such as workflow [@doi:10.1038/s41467-022-28313-9] to enable fully deep learning based design of metalloproteins. We are currently investigating integration of Metal3D combined with a classic energy based genetic algorithm based optimization. Standard machine learning models such as Random Forest have been used for classyifing metal sites as enzymatic or non-enzymatic using features computed using Rosetta and from the identity of the amino acids. Adapting our model to predict the metal and its coordination partners would allow to do a similar assessment assuming that most enzymatic metal active sites (with iron-suflur clusters as notable exception). Using standard machine learning models a prediction of the location of the metal would be much harder.  



In terms of selectivity both of our models have a clear preference for transition metals over main group metals. The only sites identified for sodium in the test set are the ones that have sidechain ligands. Many sodium and potassium sites are using backbone carbonyl coordination exclusively which is not common for zinc and those sites are therefore not detected. It has recently been demonstrated how a transition metal can be selectively bound to a protein (Choi, Tezcan, Nature 2022). This however was only achieved using non-ideal coordination which is better tolerated by some transition metals due to the different electronic configurations. It is likely impossible to train a predictor that can perfectly discriminate zinc from e.g copper and nickel sites because the coordination environments and coordination distances within the given resolution threshold are basically indistinguishable. The availability of metals inside the organism is heavily compartimentalized, therefore an approach using solely the structure will be unable to capture the full selectivity. In addition, sometimes non-native metals are used for crystallization experiments because the reaction catalyzed by an active site metal might exhibit different kinetics with different metals. The metal in the deposited structure might therefore not be the native metal. In fact, we observe that our model sometimes predicts zinc where there is a calcium modeled in the structure (EXAMPLE PDB) potentially showing another usecase for the model -> Model validation.

The HCA2 application demonstrates the utility of Metal3D for protein engineering. The kinetics of metal ion binding to proteins are complicated [@doi:10.1021/ic301645j |] and there are currently no highthroughput based approaches that could generate a dataset large enough to train a model directly on predicting K<sub>d</sub>. The data we use was obtained from a colorimetric assay predicting very high affinity of zinc[@doi:10.1021/bi00255a003]. More recent studies using ITC [@doi:10.1021/ic301645j] instead of the colorimetric assay indicate lower  K<sub>d</sub> values. In this work we chose to use the old data for consistency to estimate how well the model can recapitulate the changes in the  K<sub>d</sub> for different mutations in the first and second shell of a prototypical metalloprotein. 

Metal1D is inferior to Metal3D for the prediction of general metal ion binding sites because it produces more false positives while at the same time detecting less metal sites. This demonstrates the inherent limitation of using solely distance based features for prediction of metal location. BioMetAll which is the tool most similar to Metal1D also suffers from many false positive predictions. A possible application the authors propose is mining the protein databank for candidate metalloproteins that with relatively few mutations can be equipped with a metal ion binding site. Metal3D can be used in a similar fashion by performing rotamer sampling (e.g using a genetic algorithm) scoring each rotamer combination for its metal ion binding propensity. 


Metal3D is the first tool with angstrom level precision to predict the location of metal ion binding sites suitable for optimization of proteins. Different applications are possible: 1) Protein-Function annotation using structures predicted using Alpha Fold. 2) Integration in protein design software that performs rotamer optimization to improve the predicted metal probability near a specific point or somewhere in the protein. This could be used for engineering novel metalloproteins or metal mediated PPIs  3) Similar to detection of cryptic binding pockets one could search for cryptic metal ion binding sites in common drug targets  to  crosslink multiple faint metalbinding sites to get an inhibitive effect. 

Many of these applications allow to explore the still vastly untapped potential of proteins as large programmable multidentate metal ligands with tunable surfaces. In particular it might be easier to design protein binders using metal mediated interactions as the metal binding can overcome the entropic cost of association of two protein surfaces. Zinc is generally available in the intracellular volume and extracellurly with the total concentration of zinc in cells estimated at about 200 μM. The real available concentration of free zinc is picomolar[@doi:10.1021/cr800556u] wherefore it is crucial that design tool produces well designed sites. 

Metal3D allows to move away from using rational approaches such as the  i, i + 4 diHis motifs used for the assembly and stabilization of metalloproteins and peptides through coordination of divalent transition metal ions to a fully automated approach. [@doi:10.1126/science.8346440; @doi:10.1126/science.1648261; @doi:10.1038/nsb723]

<!-- 
Choice of resolution cutoff
Zinc The average bond length corresponds to a resolution of ∼2.2 Å. Also, the deviation increases noticeably after the resolution exceeds 2.5 Å. [@doi:10.1021/ic401072d]

Why were nucleic structures excluded? Why include all zinc sites and not just the good ones?


model choice? Why are CNNs better than equivariant approaches for this kind of work?. Recent work EquiDock uses no sidechains at all, gets ligand RMSD where only 25% are under a 2 threshold, https://arxiv.org/pdf/2202.05146.pdf Mean RMSD 8.3 A, Centroid 42.4


MIB discards all sites with less than 2 coordination partners so it will not be able to identify labile binding sites. 

Discuss 4L99 which is one of the FN for the model. Here are Lysine next to the zinc and therefore probability is low.Could be wrongly modeled (carboxlyated lysine instead of normal lysine) [@doi:10.1016/S0022-2836(02)00422-9] but some leucyl aminopeptidases also have this coordination. (check 10.1515/BC.2006.191 to find mechanism)


discuss why water molecules were not considered. considered only implicitly. Might be integrated in future but needs quality check for existing waters in training/test set 
Methods [References Dowser] that place deep waters but they do not know about metals. 


Lack in selectivity could be related to smoothing the gaussian quite a bit when training (anything >0.05) is a hit. 
Resolution of grid might be an issue 
Might be improved by improving the grid resolution to 0.5  

Discuss selectivity. E.g based on [@doi:10.1073/pnas.0906852107]. Zinc prefers tetrahedral coordination, whereas e.g copper(II) (and maybe CU(I)- check it) prefers square planar which might explain a bit lower selctivity. In the work by Tezcan the protein assemblies form different complexes with the different metals. 

General implications for the faint metal ion binding sites detected using the model: This is evident from the fact that unstructured polypeptides and folded proteins alike often form aggregates in the presence of high concentrations of transition metals.
Once the entropic cost of association is overcome, the resulting noncovalent interfaces can be optimized through additional mutations
 These nonspecific interactions could explain why free zinc concentration is tightly controlled in cellular environments; the total concentration of zinc in cells is about 200 μM, but the concentration of free zinc is only picomolar[@doi:10.1021/cr800556u].

From a practical inorganic chemical viewpoint individual proteins can be utilized as large polydentate ligands that bring along the advantage of having extensive functionalizable surfaces. From a functional perspective metals possess properties such as Lewis acidity and redox reactivity that enable them to carry out catalytic transformations not accessible by organic building blocks.

Our tool allows to move away from using rational approaches such as the  i, i + 4 diHis motifs used for the assembly and stabilization of metalloproteins and peptides through coordination of divalent transition metal ions to a more knowledgebased approach. [@doi:10.1126/science.8346440; @doi:10.1126/science.1648261; @doi:10.1038/nsb723]


Moreover, one-third of the zinc ions present in crystal structures are artifacts, merely aiding crystal formation and packing with no biological significance [@doi:10.1021/ic401072d]

 The small size of the Zn(II) cation (∼74 pm for four-coordinate and ∼88 pm for six-coordinate ion) prevents higher coordination numbers due to molecular repulsion and higher energy orbitals

Discussion with DeepSite which uses a sliding box approach and does not give individual residue scores. Therefore less useful for protein design. -->


 <!-- KD values discussion (new ITC data vs. old data by Kiefer, Fierke etc): 
 
 Old method: Enzyme-
bound zinc (E—Zn) was quantitated using the colorimetric
4-(2-pyridylazo)resorcinol (PAR) method of Hunt et al.
(1984) and measuring the absorbance at 500 nm. 4-(2-Pyridylazo)resorcinol (PAR) is a dibasic acid that forms the protonated complexes with most metal ions. It serves as a metallochromic indicator and is suitable as a chromogenic agent for the quantitative determination of over 50 elements.

removing unbound zinc by chromatography on a
PD-10 column, and measuring the protein concentration and
bound zinc concentration in the eluant using the PAR assay
(Hunt et al., 1984).
The concentration of free zinc in the
dialysis buffer was calculated from the Tris—zinc stability
constants (Dawson et al., 1986). The dissociation constant
was calculated using KaleidaGraph program with eq:
[E-ZN]/[E]tot = C/(1+Kd/[Zn]free)

Newer method: [@doi:10.1021/ic301645j]
 -->
